### PR TITLE
Rewrite the tests to remove the dependency on installed catalog

### PR DESCRIPTION
**Pull request by houshengbo**
Rewrite the tests to remove the dependency on installed catalog

This patch removed te direct call to the installed catalog and
refactor the test cases by creating local packages and actions for
the local tests.

Closes-Bug: #1164

### CODE REVIEW

File: tests/src/system/basic/WskActionSequenceTests.scala [hunk 1]
```diff
@@ -92,11 +92,31 @@ class WskActionSequenceTests
...
+
+            assetHelper.withCleaner(wsk.action, speechToTextName) {
+                val file = Some(TestUtils.getTestActionFilename("speechToText.js"))
+                (action, _) => action.create(speechToTextName, file, shared = Some(true))(wp)
+            }
+
+            assetHelper.withCleaner(wsk.action, transalteName) {
+                val file = Some(TestUtils.getTestActionFilename("translate.js"))
+                (action, _) => action.create(transalteName, file, shared = Some(true))(wp)
+            }
```
Comment (markusthoemmes) [inline tests/src/system/basic/WskActionSequenceTests.scala]: Can we come up with a different sequence of action to not having this huge dependency on Watson services and so on in the main repository?

File: tests/src/services/HeadersTests.scala [hunk 2]
```diff
@@ -122,12 +129,23 @@ class HeadersTests extends FlatSpec
...
+            val actionName = s"$packageName/helloWorld"
+            val fullActionName = s"/guest/$actionName"
+            assetHelper.withCleaner(wsk.pkg, packageName) {
+                (pkg, _) => pkg.create(packageName, shared = Some(true))
+            }
+
+            assetHelper.withCleaner(wsk.action, actionName) {
+                (action, _) => action.create(actionName, Some(TestUtils.getTestActionFilename("hello.js")))
+            }
+            val path = basePath / "namespaces" / "guest" / "actions" / "samples" / "helloWorld"
```
Comment (rabbah) [inline tests/src/services/HeadersTests.scala]: use packageName and actionName.

File: tests/src/services/HeadersTests.scala [hunk 3]
```diff
@@ -122,12 +129,23 @@ class HeadersTests extends FlatSpec
...
-    it should "respond to POST action with headers" in {
-        val path = basePath / "namespaces" / "whisk.system" / "actions" / "samples" / "helloWorld"
-        val response = pipeline(Post(url.withPath(path)) ~> addCredentials(creds)) futureValue
-
-        response.status shouldBe Accepted
-        containsHeaders(response.headers)
+    it should "respond to POST action with headers" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val packageName = "samples"
+            val actionName = s"$packageName/helloWorld"
```
Comment (rabbah) [inline tests/src/services/HeadersTests.scala]: make actionName just helloWorld.

File: tests/src/services/HeadersTests.scala [hunk 4]
```diff
@@ -122,12 +129,23 @@ class HeadersTests extends FlatSpec
...
+            val fullActionName = s"/guest/$actionName"
```
Comment (rabbah) [inline tests/src/services/HeadersTests.scala]: add package here.

**Comment by markusthoemmes** [comment]
@houshengbo please don't squash commits as we are reviewing. Just add commits associated with comments to the pull request. It makes it way easier for us to review just the delta. You can then squash when everything (initial + delta commits) is reviewed and good to go.

**Comment by houshengbo** [comment]
@markusthoemmes Thx. I am so used to combining into one commits, which is kind of habit when contributing to openstack. Will change it.

**Comment by jasonpet** [comment]
@houshengbo - if the reviews are done and this is ready you should squash the commits.

**Pull request merged**